plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java

```diff
@@ -146,6 +147,9 @@
 
     private boolean legacyHiveViewTranslation;
 
+    private boolean redirectIcebergTables;
+    private String targetCatalogForIcebergTables = "iceberg";
```

Comment:
Do we need two configs?
Maybe just
```
hive.iceberg-catalog-name
```

?

Reply:
+1
Maybe the boolean config can be just a session property to allow easy disabling for a particular query

Reply:
why would you want disable/enable this?
what use-cases are we concerned about?

(bottom line is: we should have as much flexibility as required, and no more)

Reply:
IIRC there was an agreement to use two, e.g. https://github.com/trinodb/trino/pull/4704#discussion_r477567527. But can't find previous discussion on it. I like the idea of just having one config. 

@findepi One usecase for disabling would be to easily switch back to hive if 
(1) metadata is also available in Hive and 
(2) someone would like to temporarily fallback to hive instead of iceberg if iceberg scans fails or is suboptimal for a particular query shape. Just removing `table_type=iceberg` from HMS is sufficient but not necessary for all queries and engines. 

I understand that it's still very broken to do this, but can be useful during hive->iceberg transition.

Reply:
> ) someone would like to temporarily fallback to hive instead of iceberg if iceberg scans fails or is suboptimal for a particular query shape. Just removing table_type=iceberg from HMS is sufficient but not necessary for all queries and engines.

then the redirect wouldn't happen anyway

(and user would not expect correct results, so feels like a hack)

Reply:
Anyway, with single config it's equally easy to disable redirects -- just remove the value of the config.